File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& v) {
        int c0=0,c1=0,c2=0,n=v.size();
        vector<int> sorted;
        for(int i=0;i<n;i++)
        {
            if(v[i]==0)
            {
                c0++;
            }
            else if(v[i]==1)
            {
                c1++;
            }
            else if(v[i]==2)
            {
                c2++;
            }
        }
        
        while(c0--)
        {
            sorted.push_back(0);
        }
        while(c1--)
        {
            sorted.push_back(1);
        }
        while(c2--)
        {
            sorted.push_back(2);
        }

        v.assign(sorted.begin(),sorted.end());
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void swap(int a, int b) {
        a = a ^ b;
        b = a ^ b;
        a = a ^ b;
    }

    public void sortColors(int[] nums) {
        int low = 0;
        int mid = 0;
        int n = nums.length;
        int high = n - 1;
        while (mid <= high) {
            if (nums[mid] == 0) {
                swap(nums[low], nums[mid]);
                low++;
                mid++;
            } else if (nums[mid] == 1) {
                mid++;
            } else {
                swap(nums[mid], nums[high]);
                high--;
            }
        }
    }
}